x-pack/plugins/fleet/server/services/api_keys/enrollment_api_key.ts

```diff
@@ -182,19 +176,37 @@ export async function generateEnrollmentAPIKey(
   }
 
   const name = providedKeyName ? `${providedKeyName} (${id})` : id;
-  const key = await createAPIKey(soClient, name, {
-    // Useless role to avoid to have the privilege of the user that created the key
-    'fleet-apikey-enroll': {
-      cluster: [],
-      applications: [
-        {
-          application: '.fleet',
-          privileges: ['no-privileges'],
-          resources: ['*'],
+
+  const { body: key } = await esClient.security
+    .createApiKey({
+      body: {
+        name,
+        // @ts-expect-error Metadata in api keys
+        metadata: {
```

Comment:
@nchaulet Do you know if `metadata` is the key that has to be used here or could we just use any key?